go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1171,7 +1171,7 @@ func (tsv *TabletServer) VStream(ctx context.Context, request *binlogdatapb.VStr
 	if err := tsv.sm.VerifyTarget(ctx, request.Target); err != nil {
 		return err
 	}
-	return tsv.vstreamer.Stream(ctx, request.Position, request.TableLastPKs, request.Filter, throttlerapp.VStreamerName, send)
+	return tsv.vstreamer.Stream(ctx, request.Position, request.TableLastPKs, request.Filter, throttlerapp.VStreamerName, send, request.Options)
```

Comment:
Is it worth adding vstream options to support a variadic set of options (0-N) like we recently did for lock options and (workflow) server options? Then we don't need to pass this as nil in so many places: https://github.com/vitessio/vitess/blob/main/go/vt/vtctl/workflow/server_options.go

Reply:
I prefer using strong typing unless functionally contraindicated. Especially for a grpc API. 

While updating all usages can be tedious, it also helps review the parts of code where that API is being called: https://github.com/vitessio/vitess/pull/16593#discussion_r1729537575

Reply:
Fair enough, since we already have the request object here and you've added the options to that object. Then I would ask.... why not just send the request? 
```
tsv.vstreamer.Stream(ctx, request, throttlerapp.VStreamerName, send)
```

Then we can add to the request (options, etc as we are doing here) w/o having to update all of these call sites, etc. Then you are sending no options by way of that being ZeroVal (could also mark it as optional if you prefer, where it would then be a nil pointer). And if we have things in that `VStreamRequest` that we would not want to send as part of our VStream RPC to a tablet, is that where they should be? Not a blocker or anything, by any means.